core/src/main/scala/org/apache/spark/rpc/RpcEnv.scala

```diff
@@ -264,9 +264,15 @@ object RpcTimeout {
     require(timeoutPropList.nonEmpty)
 
     // Find the first set property or use the default value with the first property
-    val foundProp = timeoutPropList.view.map(x => (x, conf.getOption(x))).filter(_._2.isDefined).
-      map(y => (y._1, y._2.get)).headOption.getOrElse(timeoutPropList.head, defaultValue)
-
+    val itr = timeoutPropList.iterator
+    var foundProp =  (timeoutPropList.head,defaultValue)
+    while (itr.hasNext && (foundProp == (timeoutPropList.head,defaultValue))){
```

Comment:
I have got no issues with that. I was anyways having second thoughts over using a "var" and a idiomatic way to loop .  I thought of keeping it as little messy as I could by avoiding another boolean variable.Anyways will make the change ASAP